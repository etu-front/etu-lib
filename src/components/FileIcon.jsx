import React from 'react'
import { Icon } from 'antd'

const icons = ['ai', 'attachment', 'audio', 'box-notes', 'csv', 'eps', 'excel']
icons.push('exe', 'flash', 'doc', 'gpres', 'gsheet', 'html', 'image', 'keynote')
icons.push('link', 'mp4', 'overlay', 'pack', 'pages', 'pdf', 'ppt', 'psd', 'rtf')
icons.push('slide', 'stypi', 'txt', 'video', 'visio', 'webex', 'word', 'xml', 'zip')

export const getIconType = fileType => {
  if (!fileType) return 'attachment'
  const type = fileType.toLowerCase()
  if (['doc', 'docx'].includes(type)) return 'word'
  if (['ppt', 'pptx'].includes(type)) return 'ppt'
  if (icons.indexOf(type) >= 0) return type
  if (['mp3', 'wav', 'ape', 'flac', 'wma'].includes(type)) return 'audio'
  if (['mov', 'ogg', 'avi', 'rmvb', 'flv', 'wmv', 'asf'].includes(type)) return 'video'
  return 'attachment'
}

const CustomIcon = Icon.createFromIconfontCN({
  scriptUrl: '//at.alicdn.com/t/font_922089_bk9l48330b8.js' // 在 iconfont.cn 上生成
})

const FileIcon = ({ style = {}, size = 50, type, ...rest }) => {
  const iconType = getIconType(type)
  return <CustomIcon type={`icon-${iconType}`} style={{ ...style, fontSize: size }} {...rest} />
}
FileIcon.displayName = 'FileIcon'
export default FileIcon
