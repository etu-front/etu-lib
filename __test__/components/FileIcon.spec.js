import React from 'react'
import FileIcon, { getIconType } from '../../src/components/FileIcon'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

const setup = () => {
  const props = {
    className: 'customClass',
    type: 'ppt',
    size: 40,
    style: { color: '#ff0000', lineHeight: '20px' },
    onClick: jest.fn()
  }
  const wrapper = shallow(<FileIcon {...props} />)
  return {
    props,
    wrapper
  }
}

describe('FileIcon', function() {
  const { props, wrapper } = setup()
  it('correct icon type', () => {
    expect(getIconType('doc')).toEqual('word')
    expect(getIconType('docx')).toEqual('word')
    expect(getIconType('pptx')).toEqual('ppt')
    expect(getIconType('ppt')).toEqual('ppt')
    expect(getIconType('mp3')).toEqual('audio')
    expect(getIconType('wma')).toEqual('audio')
    expect(getIconType('mov')).toEqual('video')
    expect(getIconType('ogg')).toEqual('video')
    expect(getIconType('blurblurblur')).toEqual('unknown')
  })

  it('props size and type', () => {
    const style = wrapper.prop('style')
    expect(style).toHaveProperty('fontSize', 40)
    expect(wrapper.prop('type')).toEqual('icon-ppt')
  })

  it('custom classname', () => {
    expect(wrapper.is('.customClass')).toEqual(true);
  })

  it('should click event be called', () => {
    wrapper.simulate('click', { key: 'Click' })
    expect(props.onClick).toBeCalled()
  })


  it('custom style', () => {
    const style = wrapper.prop('style')
    expect(style).toHaveProperty('lineHeight', '20px')
    expect(style).toHaveProperty('color', '#ff0000')
  })

  it('doc\'s icon type: icon-word', () => {
    const v = shallow(<FileIcon type='doc' />)
    expect(v.prop('type')).toEqual('icon-word')
  })

  it('render correctly', () => {
    const rendered = renderer.create(<FileIcon type={'pdf'} />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })

})
