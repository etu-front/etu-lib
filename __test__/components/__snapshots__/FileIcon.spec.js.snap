// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileIcon render correctly 1`] = `
<i
  className="anticon"
  style={
    Object {
      "fontSize": 50,
    }
  }
>
  <svg
    aria-hidden="true"
    className=""
    fill="currentColor"
    focusable="false"
    height="1em"
    width="1em"
  >
    <use
      xlinkHref="#icon-pdf"
    />
  </svg>
</i>
`;
